cmd/kubectl-direct_csi/register.go

```diff
@@ -44,28 +54,145 @@ func registerCRDs(ctx context.Context) error {
 		crdObjs = append(crdObjs, crdObj)
 	}
 
-	crdClient := utils.GetAPIExtensionsClient()
+	crdClient := utils.GetCRDClient()
 	for _, crd := range crdObjs {
 		if dryRun {
 			if err := utils.LogYAML(crd); err != nil {
 				return err
 			}
 			continue
 		}
-		res := &apiextensions.CustomResourceDefinition{}
-		if err := crdClient.RESTClient().
-			Post().
-			Resource("customresourcedefinitions").
-			VersionedParams(&metav1.CreateOptions{}, scheme.ParameterCodec).
-			Body(crd).
-			Do(ctx).
-			Into(res); err != nil {
+		var crdObj apiextensions.CustomResourceDefinition
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(crd.(*unstructured.Unstructured).Object, &crdObj); err != nil {
+			return err
+		}
+
+		existingCRD, err := crdClient.Get(ctx, crdObj.Name, metav1.GetOptions{})
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return err
+			}
+			if crdObj.Name == driveCRDName {
+				if err := setConversionWebhook(&crdObj, identity); err != nil {
```

Comment:
We should eventually make this universal, rather than per CRD type. As in, all resources should be either v1alpha1 or v1beta1 for a given release. We'll also update directcsivolumes and remove them if condition before making the next release